internal/server/errors.go

```diff
@@ -138,3 +142,19 @@ func newAPIErrorForUniqueConstraintError(ucErr data.UniqueConstraintError, msg s
 	}}
 	return apiError
 }
+
+// AuthenticationError is used to respond with a 401 Unauthorized response code.
+// Unlike internal.ErrUnauthorized, AuthenticationError includes an error message
+// in the response.
+type AuthenticationError struct {
+	// Message is sent as the api.Error.Message in the response. Message should
+	// always be a hard coded string to ensure that it does not include any
+	// sensitive data.
+	Message string
+}
+
+func (a AuthenticationError) Error() string {
+	return a.Message
+}
+
+var errMissingAuthentication = AuthenticationError{Message: "authentication is required"}
```

Comment:
Suggestions on this wording? A user is going to see this in their API client (maybe CLI, maybe connector logs, maybe UI).